schemas/io-package.json

```diff
@@ -0,0 +1,457 @@
+                        "name": {
+                            "description": "Name of TAB (multilingual object) in admin",
+                            "type": "object",
+                            "required": ["en"],
+                            "patternProperties": {
+                                "en|de|ru|pt|nl|fr|it|es|pl|zh-cn": {
+                                    "type": "string"
+                                }
+                            },
+                            "additionalProperties": false
+                        },
+                        "singleton": {
+                            "description": "If true, only one TAB for all instances will be shown.",
+                            "type": "boolean"
+                        }
+                    }
+                },
+                "controller": {
+                    "description": "Only used by js-controller",
+                    "type": "boolean"
+                },
+                "allowInit": {
+                    "description": "allow for \"scheduled\" adapter to be called \"not in the time schedule\", if settings changed or adapter started. Or allow scheduled adapter start once after configuration changed and then by schedule.",
+                    "type": "boolean"
+                },
+                "availableModes": {
+                    "description": "values for common.mode if more than one mode is possible",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "blockly": {
+                    "description": "if adapter has custom blocks for blockly (admin/blockly.js required)",
+                    "type": "boolean"
+                },
+                "connectionType": {
+                    "description": "Type of connection to device",
+                    "type": "string",
+                    "enum": ["local", "cloud"]
+                },
+                "compact": {
+                    "description": "If compact mode supported",
+                    "type": "boolean"
+                },
+                "dataFolder": {
+                    "description": "Folder relative to iobroker-data where the adapter stores the data. This folder will be backed up and restored automatically. You can use variable '%INSTANCE%' in it",
+                    "type": "string"
+                },
+                "dataSource": {
+                    "description": "How the data will be received from device",
+                    "type": "string",
+                    "enum": ["poll", "push", "assumption"]
+                },
+                "dependencies": {
+                    "description": "Array like [{\"js-controller\": \">=2.0.0\"}] that describes which ioBroker modules are required for this adapter",
+                    "type": "array",
+                    "items": {
+                        "type": ["object", "string"]
```

Comment:
is string allowed? Or only an array of Objects with a single property?

Reply:
afaik no single string is allowed.

```
[{"js-c": ">2.0.0"},
"hm-rpc"]
```

is allowed